src/lib/features/unique-connection/unique-connection-service.ts

```diff
@@ -0,0 +1,77 @@
+import type { IUnleashConfig } from '../../types/option';
+import type { IFlagResolver, IUnleashStores } from '../../types';
+import type { Logger } from '../../logger';
+import type { IUniqueConnectionStore } from './unique-connection-store-type';
+import HyperLogLog from 'hyperloglog-lite';
+import type EventEmitter from 'events';
+import { SDK_CONNECTION_ID_RECEIVED } from '../../metric-events';
+
+export class UniqueConnectionService {
+    private logger: Logger;
+
+    private uniqueConnectionStore: IUniqueConnectionStore;
+
+    private flagResolver: IFlagResolver;
+
+    private eventBus: EventEmitter;
+
+    private activeHour: number;
+
+    private hll = HyperLogLog(12);
+
+    constructor(
+        {
+            uniqueConnectionStore,
+        }: Pick<IUnleashStores, 'uniqueConnectionStore'>,
+        config: IUnleashConfig,
+    ) {
+        this.uniqueConnectionStore = uniqueConnectionStore;
+        this.logger = config.getLogger('services/unique-connection-service.ts');
+        this.flagResolver = config.flagResolver;
+        this.eventBus = config.eventBus;
+        this.activeHour = new Date().getHours();
+    }
+
+    listen() {
+        this.eventBus.on(SDK_CONNECTION_ID_RECEIVED, this.count.bind(this));
+    }
+
+    async count(connectionId: string) {
+        if (!this.flagResolver.isEnabled('uniqueSdkTracking')) return;
+        this.hll.add(HyperLogLog.hash(connectionId));
+    }
+
+    async sync(): Promise<void> {
```

Comment:
this logic will get detailed coverage. We track current and previous HyperLogLog in 2 DB rows. If we found the hour has changed we start a new current bucket and copy current to previous. We have to accommodate for multiple pods where any pod can be the first one to start a current bucket.